sdk/src/main/java/io/customer/sdk/queue/QueueRunRequest.kt

```diff
@@ -19,86 +18,67 @@ internal class QueueRunRequestImpl internal constructor(
 
-                return goToNextTask(inventory, totalNumberOfTasksToRun, null)
+            if (taskToRun == null) {
+                logger.error("Tried to get queue task with storage id: $taskStorageId but storage couldn't find it.")
+                continue
             }
-            result.isFailure -> {
-                val error = result.exceptionOrNull()
-                logger.debug("queue task $nextTaskStorageId run failed $error")
-
-                return when (error as? CustomerIOError) {
-                    is CustomerIOError.HttpRequestsPaused -> {
-                        // When HTTP requests are paused, don't increment metadata about the tasks to create inaccurate data
-                        logger.info("queue is quitting early because all HTTP requests are paused.")
 
-                        goToNextTask(emptyList(), totalNumberOfTasksToRun, null)
-                    }
-                    is CustomerIOError.BadRequest400 -> {
-                        logger.error("Received HTTP 400 response while trying to run ${nextTaskToRun.type}. 400 responses never succeed and therefore, the SDK is deleting this SDK request and not retry. Error message from API: ${error.message}, request data sent: ${nextTaskToRun.data}")
+            logger.debug("queue tasks left to run: ${tasksToRun.size}")
+            logger.debug("queue next task to run: $taskStorageId, ${taskToRun.type}, ${taskToRun.data}, ${taskToRun.runResults}")
 
-                        queueStorage.delete(nextTaskStorageId)
+            val result = runner.runTask(taskToRun)
 
-                        goToNextTask(inventory, totalNumberOfTasksToRun, lastFailedTask = nextTaskToRunInventoryItem)
-                    }
-                    else -> {
-                        val previousRunResults = nextTaskToRun.runResults
-                        val newRunResults =
-                            nextTaskToRun.runResults.copy(totalRuns = previousRunResults.totalRuns + 1)
-                        logger.debug("queue task $nextTaskStorageId, updating run history from: $previousRunResults to: $newRunResults")
-                        queueStorage.update(nextTaskStorageId, newRunResults)
+            when {
+                result.isSuccess -> {
+                    logger.debug("queue task $taskStorageId ran successfully")
+                    logger.debug("queue deleting task $taskStorageId")
+                    queueStorage.delete(taskStorageId)
+                }
 
-                        goToNextTask(inventory, totalNumberOfTasksToRun, lastFailedTask = nextTaskToRunInventoryItem)
+                result.isFailure -> {
+                    val error = result.exceptionOrNull()
+                    logger.debug("queue task $taskStorageId run failed $error")
+
+                    when (error as? CustomerIOError) {
+                        is CustomerIOError.HttpRequestsPaused, is CustomerIOError.NoHttpRequestMade -> {
```

Comment:
added `is CustomerIOError.NoHttpRequestMade` because if there is no connection, there is no point trying out every request and using resources, might as well just wait till next time.

Reply:
I don't think we should hinder getting improvement out unless we have a strong reason not to do so, it's a very small change but with a big performance improvement. I am quite comfortable making this change because `NoHttpRequestMade` is **only** being set when the network responds back with `null`. 

If you can kindly go through the code and express what makes you worry, please mention that I can respond to that concern but otherwise, renaming data types and iOS mirroring the same behavior shouldn't be the reason to block an improvement going out. The risk involved if this error shows up is it will skip an iteration unless the next time the connection goes through. 

I can request the other squad members to check out the update as well if that makes you feel more confident.

Reply:
@mrehan27 could you also go through this change when you get chance for more 👁️

Reply:
I don't see any other usage for `NoHttpRequestMade` at the moment.  So the change looks good, no reason to block unless a we find a use case that fails and can be verified by tests.

Reply:
That's my point as well, this is a serious performance concern and i am comfortable with the check and the code backs it. But i would still add more tests for it to improve confidence in later releases.